src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1762 @@
+                    LogInfo.Log($"{diagnoseMode} received: {BitConverter.ToString(romTest.ToArray())}, ret: {ret}", LogLevel.Debug);
+                    // Wait for the test to run
+                    // TODO: find the right timing, this is empirical
+                    Thread.Sleep(100);
+                    return (romTest[0] == 0) && (ret >= 0);
+                case DiagnoseMode.RAMTest:
+                    // NumTst = 0x02: RAM Test
+                    // This test is for checking RAM; 768 bytes of XRAM and 128 bytes of IDATA.
+                    // The test method used consists of saving original content, writing test data, checking test
+                    // data and finally restore original data. So, this test is non destructive.
+                    // − Parameter Length : 0,
+                    // − Result Length : 1,
+                    // − Result : 0x00 is OK,
+                    // 0xFF is Not Good. 
+                    ret = WriteCommand(CommandSet.Diagnose, singleParam);
+                    if (ret < 0) return false;
+                    // Wait for the test to run
+                    // TODO: find the right timing, this is empirical
+                    Thread.Sleep(1500);
+                    Span<byte> ramTest = stackalloc byte[1];
+                    ret = ReadResponse(CommandSet.Diagnose, ramTest);
+                    LogInfo.Log($"{diagnoseMode} received: {BitConverter.ToString(ramTest.ToArray())}, ret: {ret}", LogLevel.Debug);
+                    Thread.Sleep(100);
+                    return (ramTest[0] == 0) && (ret >= 0);
+                case DiagnoseMode.PollingTestToTarget:
+                    // NumTst = 0x04 : Polling Test to Target
+                    // This test is for checking the percentage of failure regarding response packet receiving
+                    // after polling command transmission. In this test, the PN532 sends a FeliCa polling
+                    // command packet 128 times to target. The PN532 counts the number of fails and returns
+                    // the failed number to host controller. This test doesn’t require specific system code for
+                    // target.
+                    // Polling is done with system code (0xFF, 0xFF). The baud rate used is either 212 kbps or
+                    // 424 kbps.
+                    // One polling is considered as defective after no correct polling response within 4 ms.
+                    // During this test, the analog settings used are those defined in command
+                    // RFConfiguration within the item n°7 (§7.3.1, p: 101).
+                    // − Parameter Length : 1,
+                    // − Parameter : 0x01 is 212 kbps,
+                    // 0x02 is 424 kbps.
+                    // − Result Length : 1,
+                    // − Result : Number of fails (Maximum 128). 
+                    throw new Exception($"Test {diagnoseMode} not implemented");
```

Comment:
NotImplementedException

Reply:
same below

Reply:
> LGTM after fixing the comments, please don't rebase

Made all adjustments, build fully locally, checked all ok and just pushed. So should be all ok with CI/CD build.